### PR TITLE
feat(cli): show latest blocks schema on `prefect block type inspect <slug>`

### PR DESCRIPTION
**Pull request by parkedwards**
I'd like to find the schema of a block type, given the slug.  currently `prefect block type inspect <slug>` shows some top-level metadata, like the type ID and description.  I think it makes sense to include the latest schema of the type, in a readable format

### Example
https://www.loom.com/share/b47cae0cacff4bf9907e2823ebcd60d9?sid=74f80ff0-c2dc-4598-b56b-aeb34549c1cd

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/prefect/cli/block.py [hunk 1]
```diff
@@ -72,6 +73,31 @@ def display_block_type(block_type):
...
+
+    for property_name, property_schema in properties.items():
+        if property_name in required:
+            property_schema["required"] = True
+
+        block_schema_yaml_table.add_row(
+            property_name, yaml.dump(property_schema, default_flow_style=False)
+        )
+
+    return block_schema_yaml_table
```
Comment (parkedwards) [inline src/prefect/cli/block.py:98]: https://www.loom.com/share/b47cae0cacff4bf9907e2823ebcd60d9?sid=74f80ff0-c2dc-4598-b56b-aeb34549c1cd

Comment (parkedwards) [reply src/prefect/cli/block.py:98]: we have some jsonschema `$ref` references here - originally @desertaxle and i discussed maybe parsing these to show a `prefect block type inspect <slug>` breacrumb inside, but after some trial-and-error, im realizing that:

- not all `$ref` are a block type, eg. `AwsClientParameters`, `SMTPType`, etc.
- we'd have to loop and verify each $ref to see if they're a valid block type
- _**then**_ we could hyphen-case them

i'm inclined to keep the jsonschema ref class, and we can iterate on this

Comment (parkedwards) [reply src/prefect/cli/block.py:98]: example:
image

Comment (desertaxle) [reply src/prefect/cli/block.py:98]: Here's another idea that shouldn't be too much of a lift: could you show a 'Definitions' section below 'Schema Properties'? The definitions should be included in the block schema.

Comment (parkedwards) [reply src/prefect/cli/block.py:98]: oooo yeah let me look at that

Comment (parkedwards) [reply src/prefect/cli/block.py:98]: https://www.loom.com/share/19a38ad3979643ae8ba01041449ca5a9

**Pull request merged**